extra_scripts/create_inaturalist2018_data_files.py

```diff
@@ -0,0 +1,120 @@
+"""
+This script is used to extract the iNaturalist2018 dataset disk_filelist.
+If download flag is specified, dataset will be downloaded to the 'input' path.
+Otherwise we will expect the typical iNaturalist2018 dataset structure.
+Please see: https://github.com/visipedia/inat_comp/tree/master/2018#data
+
+The script prepares training and validation data and produces
+train and validation source/label `.npy` files.
+These files will be written to the '-o' path.
+
+Usage Example:
+$ python ./create_inaturalist2018_data_files.py \
+    -i "/path/to/inaturalist2018/" \
+    -o " /output_path/to/inaturalist2018" -d
+"""
+
+import argparse
+import json
+import logging
+import numpy as np
+import sys
+from torchvision.datasets.utils import download_and_extract_archive
+
+# Initiate the logger
+FORMAT = "[%(levelname)s: %(filename)s: %(lineno)4d]: %(message)s"
+logging.basicConfig(level=logging.INFO, format=FORMAT, stream=sys.stdout)
+logger = logging.getLogger(__name__)
+
+
+def download_dataset(input: str):
+    """
+    Download the iNaturalist2018 annotations and dataset
+    """
+    logger.info("========Downloading Annotations ========")
+    TRAINING_ANNOTATIONS_URL = "https://ml-inat-competition-datasets.s3.amazonaws.com/2018/train2018.json.tar.gz"
+    VAL_ANNOTIATIONS_URL = (
+        "https://ml-inat-competition-datasets.s3.amazonaws.com/2018/val2018.json.tar.gz"
+    )
+    download_and_extract_archive(url=TRAINING_ANNOTATIONS_URL, download_root=input)
+    download_and_extract_archive(url=VAL_ANNOTIATIONS_URL, download_root=input)
+
+    logger.info("========Downloading Images. This may take awhile! ========")
+    IMAGES_URL = "https://ml-inat-competition-datasets.s3.amazonaws.com/2018/train_val2018.tar.gz"
+    download_and_extract_archive(url=IMAGES_URL, download_root=input)
+
+
+def get_images_labels_info(input_file_name: str, input: str):
+    """
+    Process the iNaturalist2018 image paths and labels for the training or validation data
+    This assumes the input contains the train2018.json and val2018.json files with relative paths to
+    training and validation images, respectively, as well as a sub-directory with all the raw images
+    """
+    data_source_path = input + input_file_name
+    with (open(data_source_path)) as ds_path:
+        data = json.load(ds_path)
+
+    images, labels = [], []
+
+    for info in data["images"]:
```

Comment:
we should add asserts `assert "images" in data, f"Please check the specified {data_source_path} as it doesn't contain 'images' in it"`